### PR TITLE
Rename getter functions to be idiomatic

### PR DESCRIPTION
**Pull request by terencechain**
**What type of PR is this?**

> Clean up

**What does this PR do? Why is it needed?**

Rename a few getter functions to make it more idiomatic for Go
More info: https://hackmd.io/CVorlf6HQpWtU5jCSPQvhg

**Which issues(s) does this PR fix?**

N/A

**Other notes for review**

N/A

### CODE REVIEW

File: beacon-chain/db/kv/blocks.go [hunk 1]
```diff
@@ -501,9 +501,9 @@ func fetchBlockRootsBySlotRange(
 	return roots, nil
 }
 
-// getBlockRootsByFilter retrieves the block roots by slot
-func getBlockRootsBySlot(ctx context.Context, tx *bolt.Tx, slot uint64) ([][]byte, error) {
-	ctx, span := trace.StartSpan(ctx, "BeaconDB.getBlockRootsBySlot")
+// blockRootsByFilter retrieves the block roots by slot
```
Comment (rkapka) [inline beacon-chain/db/kv/blocks.go:504]:
```suggestion
// blockRootsBySlot retrieves the block roots by slot
```

File: beacon-chain/p2p/service.go [hunk 2]
```diff
@@ -83,7 +83,7 @@ type Service struct {
 }
 
 // NewService initializes a new p2p service compatible with shared.Service interface. No
-// connections are made until the Start function is called during the service registry startup.
+// connections are made until the New function is called during the service registry startup.
```
Comment (rkapka) [inline beacon-chain/p2p/service.go:86]: Are you sure this change is correct?

File: beacon-chain/p2p/service.go [hunk 3]
```diff
@@ -83,7 +83,7 @@ type Service struct {
...
 func NewService(ctx context.Context, cfg *Config) (*Service, error) {
```
Comment (rkapka) [inline beacon-chain/p2p/service.go:87]: I would prefer `New`. Similarly for other services.

File: beacon-chain/rpc/beacon/attestations.go [hunk 4]
```diff
@@ -103,7 +103,7 @@ func (bs *Server) ListAttestations(
 }
 
 // ListIndexedAttestations retrieves indexed attestations by block root.
-// IndexedAttestationsForEpoch are sorted by data slot by default. Start-end epoch
+// IndexedAttestationsForEpoch are sorted by data slot by default. New-end epoch
```
Comment (rkapka) [inline beacon-chain/rpc/beacon/attestations.go:106]: Something is not right here.

File: shared/slotutil/slotticker.go [hunk 5]
```diff
@@ -38,8 +38,8 @@ func (s *SlotTicker) Done() {
 	}()
 }
 
-// GetSlotTicker is the constructor for SlotTicker.
-func GetSlotTicker(genesisTime time.Time, secondsPerSlot uint64) *SlotTicker {
+// New starts and returns a new SlotTicker instance.
+func New(genesisTime time.Time, secondsPerSlot uint64) *SlotTicker {
```
Comment (rkapka) [inline shared/slotutil/slotticker.go:42]: I disagree with this change - `slotutil.New` does not convey information about what is being created. I would prefer `slotutil.NewSlotTicker`.

File: endtoend/helpers/epochTimer.go [hunk 6]
```diff
@@ -30,8 +30,8 @@ func (s *EpochTicker) Done() {
 	}()
 }
 
-// GetEpochTicker is the constructor for EpochTicker.
-func GetEpochTicker(genesisTime time.Time, secondsPerEpoch uint64) *EpochTicker {
+// New starts the EpochTicker.
+func New(genesisTime time.Time, secondsPerEpoch uint64) *EpochTicker {
```
Comment (rkapka) [inline endtoend/helpers/epochTimer.go:34]: I disagree with this change - `helpers.New` does not convey information about what is being created. I would prefer `helpers.NewEpochTicker`.

File: shared/slotutil/slotticker.go [hunk 7]
```diff
@@ -51,9 +51,9 @@ func GetSlotTicker(genesisTime time.Time, secondsPerSlot uint64) *SlotTicker {
 	return ticker
 }
 
-// GetSlotTickerWithOffset is a constructor for SlotTicker that allows a offset of time from genesis,
+// NewWithOffset starts and returns a SlotTicker instance that allows a offset of time from genesis,
 // entering a offset greater than secondsPerSlot is not allowed.
-func GetSlotTickerWithOffset(genesisTime time.Time, offset time.Duration, secondsPerSlot uint64) *SlotTicker {
+func NewWithOffset(genesisTime time.Time, offset time.Duration, secondsPerSlot uint64) *SlotTicker {
```
Comment (rkapka) [inline shared/slotutil/slotticker.go:56]: We don't know what is that thing with offset. I would prefer `NewSlotTickerWithOffset`.

**Review by rkapka — CHANGES_REQUESTED** [review]
I would like to propose the following standard for naming constructors.

If the package name describes the object being created, then the constructor name should be `New`, to avoid duplication. Example: `service.New` vs `service.NewSevice` (the function returns a `Service` struct). Otherwise, the name should be `NewXyz`. Example: `helpers.NewSlotTicker` vs `helpers.New`. Seeing `helpers.New` does not tell you anything about what's being created.

File: beacon-chain/p2p/service.go (hunk 3 above)
Comment (terencechain) [reply beacon-chain/p2p/service.go:87]: Yes. But let's do it in another PR

**Review by rkapka — APPROVED** [review]

**Pull request merged**